source/Calamari.Aws/Kubernetes/Discovery/AwsAuthenticationDetails.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Collections.Generic;
+using Amazon.Runtime;
+using Calamari.Common.Features.Discovery;
+
+namespace Calamari.Kubernetes.Aws
+{
+    public class AwsAuthenticationDetails : ITargetDiscoveryAuthenticationDetails
+    {
+        const string DefaultSessionName = "OctopusKubernetesClusterDiscovery";
+        public AWSCredentials ToCredentials()
+        {
+            AWSCredentials account;
+            if (Credentials.Type == "account")
+            {
+                account = new BasicAWSCredentials(Credentials.Account.AccessKey, Credentials.Account.SecretKey);
+            }
+            else
+            {
+                try
+                {
+                    // If not currently running on an EC2 instance,
+                    // this will throw an exception.
+                    account = new InstanceProfileAWSCredentials();
+                }
+                catch
+                {
+                    // The last attempt is trying to use Environment Variables.
+                    account = new EnvironmentVariablesAWSCredentials();
```

Comment:
As per our discussion, if this constructor can throw we should catch the possible exceptions here, and make sure appropriate error messages are generated.

Otherwise, let's make sure that the error case where no env vars exist is handled at the point at which it surfaces.

Reply:
Done 👍